### PR TITLE
add resource retriever and embed reader

### PR DESCRIPTION
**Pull request by gregLibert**

### CODE REVIEW

File: pkg/resource/embed.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	if err == fs.ErrNotExist {
+		return nil, ErrNotExist
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return content, nil
+}
```
Comment (ASAPSegfault) [inline pkg/resource/embed.go:24]: Why not closing the file ? With a defer, like that it would handle all use cases with 1 line

Comment (ASAPSegfault) [reply pkg/resource/embed.go]: Cool!

**Review by ASAPSegfault — APPROVED** [review]

**Review by 0xMazout — APPROVED** [review]
LGTM :)

**Comment by SlnPons** [comment]
@gregLibert how can I test this issue please?

**Pull request merged**